src/components/app/App.js

```diff
@@ -1,11 +1,33 @@
-import React from 'react';
+import React, {createContext, useReducer} from 'react';
 import '../../styles/App.scss';
+import  Home from '../home/Home';
+import CenterContainer from '../votingcenter/CenterContainer'
 
+const initialState = {
+  address: null
+}
+
+const reducer = (state, action) => {
+  switch(action.type){
+    case "submitAddress":
+      return {...state,  address: action.payload}
+    default:
+      return state;
+  }
+}
```

Comment:
useReducer behaves similarly to the reducer in redux. here we're using it to manage a state object that is shared across the app via useContext.